src/borg/repository.py

```diff
@@ -173,6 +173,21 @@ def load_key(self):
         keydata = self.config.get('repository', 'key')
         return keydata.encode('utf-8')  # remote repo: msgpack issue #99, returning bytes
 
+    def get_free_nonce(self):
+        nonce_path = os.path.join(self.path, 'nonce')
+        if os.path.exists(nonce_path):
```

Comment:
how about instead exists() + open() just catching the exception if it does not open, because file is missing?